services/compute/virtualmachine/virtualmachine.go

```diff
@@ -607,6 +607,7 @@ func (c *client) getVirtualMachineStorageProfileOsDisk(d *wssdcloudcompute.Disk)
 		}
 	}
 	return &compute.OSDisk{
+		Name:        &d.Diskname,
```

Comment:
Why cant we just use VHD? Why do we need the name? Since the vallues are the same

Reply:
The idea here is to remove it from the VHD.URI because it doesn't makes sense to have the name there. I wanted to be sure that it is not affecting any other team before doing it. So will do it as part of a seperate PR later and run the tests.